### ISSUE TITLE
Procedure

### ISSUE THREAD

**Issue by mbeaufils** [opengeospatial/Geotech#34]
As defined in OMS : 

1)	Procedure is an abstract concept that is then further specialized in the various procedure types defined in this document. All share the commonality of describing a defined series of steps to a specific purpose.
2)	The term process that was used in ISO 19156:2011 was purposely dropped in this version to avoid unnecessary confusion between the terms procedure and process.

**Comment by LucasoilCloud** [comment]
I agree that the procedure has to be recorded for each observation. Maybe for simplifing we could only indicate the standard used for the acquisition of the observation. For example ISO 22476-3 for SPT test. 

For the fun, I have checked the wikipedia page for the SPT test and it is indicated 'The test **procedure** is described in ISO 22476-3' so implicitly that means that if we indicate the standard, we have the procedure.

**Comment by mbeaufils** [comment]
Ideally the description of the procedure should be freely accessible.

**Comment by LucasoilCloud** [comment]
Why not. So the procedure has to be considered as a 'string' type property of an observation. The value can be a standard reference or anything else.

**Comment by dponti** [comment]
All observations result from some procedure, but I think we need to be careful not to create a schema/model that requires the reporting of a procedure for every observation. In working/using legacy data, sometimes information on the process that leads to a result is lost or was never documented. While not having this information may compromise or limit the utility of the results, depending on the use case, but it doesn't necessarily render the result useless.

**Comment by mbeaufils** [comment]
An observation is associated to 1 ObservingProcedure

![image](https://user-images.githubusercontent.com/17067226/161117263-a9ba2708-1c5b-40d3-b4b0-8f432e0fc7c7.png)

**Comment by mbeaufils** [comment]
@dponti and other people that would like to propose a definition of a process.
A proposal from the OGC standard SensorML exists. It worth been considered: http://docs.ogc.org/is/12-000r2/12-000r2.html

![image](https://user-images.githubusercontent.com/17067226/165127238-58c4f689-b95a-4536-8066-4fb742053d78.png)

**Comment by dponti** [comment]
Below is a proposed start at a UML model for a Geotechnical Process (GT_Process). This object is intended to be the object that is referended by the Observation's procedure property. Following I'll give some encoding examples. The intent of this model is to accommodate pretty much any kind of chain of complex procedural observations that are then used to determine the observation's "result". The process can contain a chain of "procedures" each one of which can have any number of nested "procedural observations". Metadata is handled by the parameter property, which uses OM's NamedValue object.

![Procedure Perspective - Page 1 (1)](https://user-images.githubusercontent.com/26747679/165738678-15dfa014-0071-45b7-98fe-f7d01915166c.png)

**Comment by dponti** [comment]
Attached are encoding examples of GT_Process for a Lugeon test procedure and a Partlcle Size test procedure using the above model in JSON notation (my apologies for any screw ups as my JSON skills are poor). I'm more comfortable in XML but this might be more readable. The extra nesting is due to conformance with GML's object-property rule.
[LugeionExample.txt](https://github.com/opengeospatial/Geotech/files/8582162/LugeionExample.txt)
[ParticleSizeExample.txt](https://github.com/opengeospatial/Geotech/files/8582164/ParticleSizeExample.txt)

**Comment by dponti** [comment]
The below diagram lists the test (observation) procedures currently supported in DIGGS. The procedural attributes are (mercifully) omitted here. Attached, however is a spreadsheet that lists the observed properties for observations (sheet Definitions) and the procedures associated with these result properties (sheet AssociatedElements)
![Procedure Perspective - Page 1 (3)](https://user-images.githubusercontent.com/26747679/166865204-9dcd734c-2862-47b7-a1cc-721e098bfbfb.png)
[DIGGSTestPropertyDefinitions.xlsx](https://github.com/opengeospatial/Geotech/files/8628765/DIGGSTestPropertyDefinitions.xlsx)

**Comment by mbeaufils** [comment]
Thanks @dponti. Also I like the listing of the Properties per Procedure.

Is this ObservedProperties (meaning the Procedure ALWAYS address those properties) or ObservableProperties (meaning the Procedure CAN address those properties)?
